### PR TITLE
Add changelog for Enterprise 3.8.0

### LINKED ISSUE #2354: Sensu Enterprise 3.8 Docs
### Description

Sensu Enterprise 3.8.0 is waiting in the wings, we need to update docs for it.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Additional context

We think this is the last ever Sensu Enterprise release, but want to have a changelog etc to go with it.

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds changelog entry for Sensu Enterprise 3.8.0.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2354

## Review Instructions
I doubt if I've captured the correct information for 3.8.0. For example:
- The link https://github.com/sensu/sensu-eol in https://github.com/sensu/sensu-enterprise/pull/445 is a 404 for me -- I can see Justin has some PRs against it via #classic, so maybe it's just not public yet.
- The [Enterprise repo changelog](https://github.com/sensu/sensu-enterprise/blob/master/CHANGELOG.md#380---2020-04-08) says 3.8.0 is based on Core 1.10.0, but the Core repo changelog doesn't list version 1.10.0.

### CODE REVIEW

**Comment by hillaryfraley** [comment]
Cameron is going to sort out access to /sensu-eol repo, where I will find the info I need to update the 3.8.0 changelog.

File: content/sensu-enterprise/3.8/changelog.md [hunk 1]
```diff
@@ -60,6 +61,22 @@ _NOTE: Sensu Enterprise is built on Sensu Core. Sensu Core changes are documente
...
 
+## Enterprise 3.8.0 Release Notes {#enterprise-v3-8-0}
+
+**April 8, 2020** &mdash; Sensu Enterprise version 3.8.0 has been
+released and is available for immediate download. Please note the
+following improvements:
+
+### CHANGES {#enterprise-v3-8-0-changes}
+
+Built on Sensu Core 1.10.0:
```
Comment (cwjohnston) [inline content/sensu-enterprise/3.8/changelog.md]:
```suggestion
Built on Sensu Core 1.10.0 (internal release):
```

File: content/sensu-enterprise/3.8/changelog.md [hunk 2]
```diff
@@ -60,6 +61,22 @@ _NOTE: Sensu Enterprise is built on Sensu Core. Sensu Core changes are documente
...
+
+- **NEW**: Updates the gem name from `sensu` to `sensu-eol`, which is built from https://github.com/sensu/sensu-eol.
```
Comment (cwjohnston) [inline content/sensu-enterprise/3.8/changelog.md]:
```suggestion
```

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-enterprise/3.8/changelog.md (hunk 2 above)
Comment (cwjohnston) [reply content/sensu-enterprise/3.8/changelog.md]: This isn't really a user-facing detail, so we should just describe it above as an internal release.

File: content/sensu-enterprise/3.8/changelog.md [hunk 3]
```diff
@@ -60,6 +61,21 @@ _NOTE: Sensu Enterprise is built on Sensu Core. Sensu Core changes are documente
...
+## Enterprise 3.8.0 Release Notes {#enterprise-v3-8-0}
+
+**April 8, 2020** &mdash; Sensu Enterprise version 3.8.0 has been
+released and is available for immediate download. Please note the
+following improvements:
+
+### CHANGES {#enterprise-v3-8-0-changes}
+
+Built on Sensu Core 1.10.0 (internal release):
+
```
Comment (hillaryfraley) [inline content/sensu-enterprise/3.8/changelog.md:73]:
```suggestion
```

**Pull request merged**